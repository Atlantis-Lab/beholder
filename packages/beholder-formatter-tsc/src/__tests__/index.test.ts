import path from 'path'
import fs from 'fs'
import { format } from '../'

const loadFixtures = name =>
  fs.readFileSync(path.join(__dirname, 'fixtures', `${name}.txt`)).toString()

describe('format', () => {
  it('should match base snapshot', async () => {
    const result = await format(loadFixtures('base'))

    expect(result).toMatchSnapshot()
  })
})
