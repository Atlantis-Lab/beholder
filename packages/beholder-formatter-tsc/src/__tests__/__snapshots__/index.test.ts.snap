// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format should match base snapshot 1`] = `
Object {
  "annotations": Array [
    Object {
      "annotation_level": "failure",
      "end_column": "16",
      "end_line": "1",
      "message": " Cannot find module 'fs'.",
      "path": "packages/test/src/invalid.ts",
      "raw_details": "( error TS2307):  Cannot find module 'fs'.",
      "start_column": "16",
      "start_line": "1",
      "title": " error TS2307",
    },
    Object {
      "annotation_level": "failure",
      "end_column": "13",
      "end_line": "4",
      "message": " Cannot find name 'some'.",
      "path": "packages/test/src/invalid.ts",
      "raw_details": "( error TS2304):  Cannot find name 'some'.",
      "start_column": "13",
      "start_line": "4",
      "title": " error TS2304",
    },
  ],
  "summary": "Found 2 errors",
  "title": "TypeScript",
}
`;
